lib/ansible/utils/vars.py

```diff
@@ -230,33 +230,42 @@ def load_options_vars(version):
     return options_vars
 
 
-def isidentifier(ident):
-    """
-    Determines, if string is valid Python identifier using the ast module.
+def isidentifier_PY3(ident):
+    """Determine if string is valid Python identifier for Python 3.
     Originally posted at: http://stackoverflow.com/a/29586366
     """
-
     if not isinstance(ident, string_types):
         return False
 
-    try:
-        root = ast.parse(ident)
-    except SyntaxError:
+    if not ident.isidentifier():
         return False
 
-    if not isinstance(root, ast.Module):
+    if keyword.iskeyword(ident):
         return False
 
-    if len(root.body) != 1:
+    return True
+
+
+def isidentifier_PY2(ident):
+    """Determine if string is valid Python identifier for Python 2.
+    Originally posted at: http://stackoverflow.com/a/29586366
+    """
+    if not isinstance(ident, string_types):
         return False
 
-    if not isinstance(root.body[0], ast.Expr):
+    if not ident:
         return False
 
-    if not isinstance(root.body[0].value, ast.Name):
+    if C.INVALID_VARIABLE_NAMES.search(ident):
         return False
 
-    if root.body[0].value.id != ident:
+    if keyword.iskeyword(ident) or ident in ("True", "False", "None"):
```

Comment:
You can make the list a frozenset at the toplevel of the file to speed this up slightly..

Reply:
Done.